src/renderer/PresetsBar.tsx

```diff
@@ -0,0 +1,215 @@
+    setIsEditMode(true);
+  };
+
+  const handleDeleteClicked = () => {
+    handleDelete(value);
+  };
+
+  const handleInputChange = (newValue: string) => {
+    handleChange(newValue);
+    setIsEditMode(false);
+  };
+
+  const handleEscape = () => {
+    setIsEditMode(false);
+  };
+
+  return (
+    <>
+      {isEditMode ? (
+        <TextInput
+          ref={editModeRef}
+          value={value}
+          ariaLabel="Edit Preset Name"
+          isDisabled={false}
+          handleChange={handleInputChange}
+          handleEscape={handleEscape}
+          updateOnSubmitOnly
+        />
+      ) : (
+        <>
+          {value}
+          <div className="row icons">
+            <IconButton
+              icon={IconName.EDIT}
+              handleClick={handleEditClicked}
+              isDisabled={isDisabled}
+            />
+            <IconButton
+              icon={IconName.DELETE}
+              handleClick={handleDeleteClicked}
+              isDisabled={isDisabled}
+            />
+          </div>
+        </>
+      )}
+    </>
+  );
+};
+
+const PresetsBar = () => {
+  const { globalError } = useAquaContext();
+
+  const [presetName, setPresetName] = useState<string>('');
+  const [selectedPresetName, setSelectedPresetName] = useState<
+    string | undefined
+  >(undefined);
+
+  // TODO: Fetch default presets and custom presets from storage
+  const [presetNames, setPresetNames] = useState<string[]>([
+    'Airpods Pro to IEF Neutral AutoEQ',
+    'Airpods to Harman AE OE 2018 AutoEQ',
+    '7-1 Surround Sound Speaker Setup',
+    'Bass Boost',
+    'Bass and Treble Boost',
+    'Classic',
+    'Dance',
+    'Rock',
+    'Vocal Boost',
+    'Microphone Compensation',
+    'MORE BASS',
+    'Surround effect',
+    'Indie playlist',
+    'Pop',
+    'Hip-hop',
+    'R&B',
+    'Equalize for shrill noise',
+    'Jazz',
+  ]);
+
+  const handleCreatePreset = (prev: string[]) => {
+    if (prev.indexOf(presetName) !== -1) {
+      // TODO: overwrite preset values for the existing preset
+    } else {
+      // TODO: add code for creating a preset
+      setSelectedPresetName(presetName);
+      const newPresets = [...prev, presetName].sort();
+      setPresetNames(newPresets);
+    }
+  };
+
+  const handleLoadPreset = () => {
+    // TODO: add code for loading a preset
+  };
+
+  const handleChangeNewPresetName = (newValue: string) => {
+    setPresetName(newValue);
+
+    // TODO: If the preset name is an existing one, we should set the selected one to be that so we can load it
+    setSelectedPresetName(undefined);
+  };
+
+  const handleChangeSelectedPreset = (newValue: string) => {
+    setPresetName(newValue);
+    setSelectedPresetName(newValue);
+  };
+
+  const options: IOptionEntry[] = useMemo(() => {
+    const handleEditExistingPresetName = (
+      oldValue: string,
+      newValue: string
+    ) => {
+      // TODO: improve DS to help check if the newValue is an existing preset or not
```

Comment:
From our last meeting, we've decided not to care. Add a comment here about that.

Reply:
I thought we still needed to figure out if the preset would overwrite vs create a new preset? The UI would still need to identify which use case should occur so it can update the state correctly no?